src/Stratis.Bitcoin.Tests/P2P/PeerConnectorTests.cs

```diff
@@ -367,6 +367,7 @@ private static bool ConnectToPeer(PeerAddressManager peerAddressManager, Mock<IN
                     It.IsAny<NetworkPeerConnectionParameters>(), It.IsAny<NetworkPeerDisposer>())).Returns(Task.FromResult(networkPeer.Object));
             connectionManagerSettingsExisting.AddNode.Add(endpointNode);
             connectionManagerSettingsExisting.Connect.Add(endpointNode);
+            AddMissingPeersToConnectionManager(peerConnector);
```

Comment:
There should be no need for this, just check whether you want to return true or false from this method and if true, just add it to the peer collection

Reply:
I agree - but the problem is "just add it to the peer collection"
because the collection is immutable and private         
private IReadOnlyNetworkPeerCollection connectedPeers;

Reply:
@fenix2222 go look at ConnectionManager there is a `AddConnectedPeer` method on it ;)

Reply:
@fassadlr @fenix2222 != @thefazzer , otherwise you are perfectly right ;)

Reply:
@Aprogiena hahaha tag fail 👎